pennylane/devices/preprocess.py

```diff
@@ -196,6 +196,11 @@ def mid_circuit_measurements(
     if mcm_method is None:
         mcm_method = "one-shot" if tape.shots else "deferred"
 
+    if any(is_conditional_mcm(op) for op in tape.operations):
```

Comment:
A few questions on this one:

* Any performance concerns with introducing this `O(n)` call?
* Would this occur with every execution, or just the first time?
* If every time, would it be worth moving this to a separate function with an LRU decorator to immediately raise on subsequent calls if the tape has a hit?

Reply:
> Any performance concerns with introducing this O(n) call?

Yes - every time we iterate over the tape, there's a performance concern, especially as we allowing prioritize larger tapes.

> Would this occur with every execution, or just the first time?

I don't think we have any caching for preprocess, I believe it would occur every time.

> If every time, would it be worth moving this to a separate function with an LRU decorator to immediately raise on subsequent calls if the tape has a hit?

That's an interesting idea!

Reply:
Discussed offline with @mudit2812. We can't move it to `cond` because we want to allow the creation of these operators, and then provide a transform that - in some circumstances - makes them executable (currently a rough draft, [here](https://github.com/PennyLaneAI/pennylane/pull/7037)). So this could go into the stopping condition in `decompose` to avoid an additional iteration over the tape, or we can add an LRU decorator. I'll take another look at it 😄

Reply:
Added a cache :)

Reply:
Sounds good, thanks!